### PR TITLE
Adds some FIRE_PROOF checks

### LINKED ISSUE #42555: Fireproofed jumpsuit from chill burns still
Issue reported from  Round ID: 100954 (/tg/Station Terry [EU] [100% LAG FREE])

Reporting client version: 512

End of round, my fireproofed jumpsuit burnt off

### PR DESCRIPTION
**Pull request by 4dplanner**
## About The Pull Request

Checks FIRE_PROOF when igniting an object, and when continuing to burn it
## Why It's Good For The Game
Currently clothing you wear sets on fire, and only doesn't take damage because it's fireproof for the initial hit.

## Changelog
:cl:
fix: fireproofing works properly
/:cl:

fixes #42555
(for real this time)

### CODE REVIEW

**Review by Cyberboss — DISMISSED** [review]

File: code/controllers/subsystem/fire_burning.dm [hunk 1]
```diff
@@ -27,9 +27,10 @@ SUBSYSTEM_DEF(fire_burning)
 				return
 			continue
 
-		if(O.resistance_flags & ON_FIRE)
+		if((O.resistance_flags & ON_FIRE) && !(O.resistance_flags & FIRE_PROOF))
 			O.take_damage(20, BURN, "fire", 0)
 		else
+			O.resistance_flags &= ~ON_FIRE //in case we fireproof while burning
```
Comment (ShizCalev) [inline code/controllers/subsystem/fire_burning.dm]: this is already handled at object level.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/controllers/subsystem/fire_burning.dm (hunk 1 above)
Comment (4dplanner) [reply code/controllers/subsystem/fire_burning.dm]: Or are you saying I should call extinguish() here

Comment (ShizCalev) [reply code/controllers/subsystem/fire_burning.dm]: Nothing should be setting that flag directly as it's supposed to always be handled by extinguished(). If anything is actually doing it, that's a bug that needs to be fixed.

Comment (4dplanner) [reply code/controllers/subsystem/fire_burning.dm]: Gotcha

Comment (4dplanner) [reply code/controllers/subsystem/fire_burning.dm]: I'll have to clear this up tomorrow, though, it's late uk time.

File: code/controllers/subsystem/fire_burning.dm [hunk 2]
```diff
@@ -27,9 +27,10 @@ SUBSYSTEM_DEF(fire_burning)
...
 			processing -= O
```
Comment (ShizCalev) [inline code/controllers/subsystem/fire_burning.dm]: actually, this is probably why things sometimes don't have their burning overlays cleared out. do a call for O.extinguished() instead.

**Review by ShizCalev — DISMISSED** [review]

File: code/controllers/subsystem/fire_burning.dm (hunk 2 above)
Comment (4dplanner) [reply code/controllers/subsystem/fire_burning.dm]: For context: 
`/obj/proc/extinguish()
	if(resistance_flags & ON_FIRE)
		resistance_flags &= ~ON_FIRE
		cut_overlay(GLOB.fire_overlay, TRUE)
		SSfire_burning.processing -= src
`
Anytime ON_FIRE gets removed, we should already have removed the item from processing.

File: code/controllers/subsystem/fire_burning.dm [hunk 3]
```diff
@@ -27,10 +27,11 @@ SUBSYSTEM_DEF(fire_burning)
 				return
 			continue
 
-		if(O.resistance_flags & ON_FIRE)
```
Comment (ShizCalev) [inline code/controllers/subsystem/fire_burning.dm]: where was the redundancy for this?

Comment (4dplanner) [reply code/controllers/subsystem/fire_burning.dm]: Just realised it's in there for if something is extinguished mid-run, will put it back

File: code/controllers/subsystem/fire_burning.dm [hunk 4]
```diff
@@ -27,10 +27,12 @@ SUBSYSTEM_DEF(fire_burning)
...
-			O.take_damage(20, BURN, "fire", 0)
-		else
-			processing -= O
```
Comment (4dplanner) [inline code/controllers/subsystem/fire_burning.dm]: I think this particular line wasn't doing anything, though, as extinguish() already removes O from processing.

**Comment by ExcessiveUseOfCobblestone** [comment]
Granny PR

**Comment by 4dplanner** [comment]
It's not stale, it's ignored by the maintainers thanks

File: code/controllers/subsystem/fire_burning.dm (hunk 4 above)
Comment (ExcessiveUseOfCobblestone) [reply code/controllers/subsystem/fire_burning.dm]: correct

**Comment by ExcessiveUseOfCobblestone** [comment]
resolve outdated reviews evil lady

**Pull request merged**